lib/components/narrative/narrative-itineraries.js

```diff
@@ -164,24 +231,90 @@ class NarrativeItineraries extends Component {
+        if (
+          !duplicateItin.allStartTimes.find(
+            (time) => time.time === cur.startTime
+          )
+        ) {
+          duplicateItin.allStartTimes.push({
+            arrival: cur.endTime,
+            realtime: firstTransitLegIsRealtime(cur),
+            time: cur.startTime
+          })
         }
-        // Ensure that the lowest start time is always first
-        duplicateItin.allStartTimes.add(cur.startTime)
+
+        // Some legs will be the same, but have a different route
+        // This map catches those and stores the alternate routes so they can be displayed
+        duplicateItin.legs = duplicateItin.legs.map((leg, index) => {
+          const newLeg = clone(leg)
+          if (leg?.routeId !== cur.legs[index]?.routeId) {
+            if (!newLeg.alternateRoutes) {
+              newLeg.alternateRoutes = {}
+            }
+            const { routeId } = cur.legs?.[index]
+            newLeg.alternateRoutes[routeId] = {
+              // We save the entire leg to the alternateRoutes object so in
+              // the future, we can draw the leg on the map as an alternate route
+              ...cur.legs?.[index]
+            }
+          }
+          return newLeg
+        })
       }
       return updatedItineraries
     }, [])
 
-    const showRealtimeAnnotation =
-      realtimeEffects.isAffectedByRealtimeData &&
-      (realtimeEffects.exceedsThreshold || realtimeEffects.routesDiffer)
+    // This loop determines if an itinerary uses a single or multiple modes
+    const groupedMergedItineraries = mergedItineraries.reduce(
+      (prev, cur) => {
+        // Create a clone of our buckets
+        const modeItinMap = clone(prev)
+        // We generate a mode string description as this handles
+        // a lot of the itinerary processing work for us
+        const modeString = getItineraryDescription({
+          combineTransitModes: groupTransitModes,
+          intl,
+          itinerary: cur
+        })
+
+        // We determine which "bucket" to put this itinerary into based on what
+        // is currently a hack hack hack: looking at the presence of a space
+        // in the mode string.
+        //
+        // Another approach may be cleaner, but this includes the least overhead
+        const modeContainer =
+          modeString.indexOf(' ') > -1 ? modeItinMap.multi : modeItinMap.single
+
+        // Now that we know the mode container to place our itinerary in, we do so
+        if (!modeContainer[modeString]) modeContainer[modeString] = []
+        modeContainer[modeString].push(cur)
+        return modeItinMap
+      },
+      { multi: {}, single: {} }
+    )
 
     return (
-      <S.NarrativeItinerariesContainer className="options itinerary">
+      <S.NarrativeItinerariesContainer
+        className={`options itinerary ${
+          customBatchUiBackground && !itineraryIsExpanded && 'base-color-bg'
+        }`}
+      >
         <NarrativeItinerariesHeader
```

Comment:
showHeaderText seems to always be false, so the heading text never renders. If it does, should be white instead of black.

Reply:
This is configurable, and I believe the color is set via config